test/components/preprocessors/test_markdown_header_splitter.py

```diff
@@ -0,0 +1,266 @@
+        "## Header 1.2\n"
+        "### Subheader 1.2.1\n"
+        "Content under header 1.2.1.\n"
+        "### Subheader 1.2.2\n"
+        "Content under header 1.2.2.\n"
+        "### Subheader 1.2.3\n"
+        "Content under header 1.2.3."
+    )
+
+
+# Basic splitting and structure
+def test_basic_split(sample_text):
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content=sample_text)]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+
+    # Should split into all headers with content
+    headers = [doc.meta["header"] for doc in split_docs]
+    assert "Header 1" in headers
+    assert "Subheader 1.1.1" in headers
+    assert "Subheader 1.2.1" in headers
+    assert "Subheader 1.2.2" in headers
+    assert "Subheader 1.2.3" in headers
+
+    # Check that content is present and correct
+    for doc in split_docs:
+        assert doc.content.startswith("#") or doc.content.startswith("##") or doc.content.startswith("###")
+        assert doc.meta.get("header") is not None
+
+
+def test_split_parentheaders(sample_text):
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content=sample_text), Document(content="# H1\n## H2\n### H3\nContent")]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+    # Check parentheaders for both a deep subheader and a simple one
+    subheader_doc = next(doc for doc in split_docs if doc.meta["header"] == "Subheader 1.2.2")
+    assert "Header 1" in subheader_doc.meta["parentheaders"]
+    assert "Header 1.2" in subheader_doc.meta["parentheaders"]
+    h3_doc = next((doc for doc in split_docs if doc.meta["header"] == "H3"), None)
+    if h3_doc:
+        assert h3_doc.meta["parentheaders"] == ["H1", "H2"]
+
+
+def test_split_no_headers():
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content="No headers here."), Document(content="Just some text without headers.")]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+    # Should return one doc per input, header is None
+    assert len(split_docs) == 2
+    for doc in split_docs:
+        assert doc.meta["header"] is None
+
+
+def test_split_multiple_documents(sample_text):
+    splitter = MarkdownHeaderSplitter()
+    docs = [
+        Document(content=sample_text),
+        Document(content="# Another Header\nSome content."),
+        Document(content="# H1\nA"),
+        Document(content="# H2\nB"),
+    ]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+    headers = {doc.meta["header"] for doc in split_docs}
+    assert {"Another Header", "H1", "H2"}.issubset(headers)
+
+
+def test_split_only_headers():
+    text = "# H1\n# H2\n# H3"
+    splitter = MarkdownHeaderSplitter()
+    docs = [Document(content=text)]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+    # Should not create chunks for headers with no content
+    assert len(split_docs) == 0
```

Comment:
Just wanting to understand here. Why is the output not three documents with each document's content just being the header? 

Just from the name of the component I would have expected to still split on header even if the content between is empty.